### PR TITLE
Add parameter to force using GlobalMemory launch mechanism using HIP

### PR DESCRIPTION
**Pull request by Rombur**
Add a parameter that users can use to force a kernel to be launched using `GlobalMemory`. This is sometimes necessary due to compiler bugs in ROCm 5.3 and ROCm 5.4 see https://github.com/kokkos/kokkos/issues/5606 and https://github.com/kokkos/kokkos/issues/5690

cc: @arghdos

### CODE REVIEW

File: core/unit_test/TestViewAPI.hpp [hunk 1]
```diff
@@ -1542,6 +1542,7 @@ class TestViewAPI {
   }
 
   static void run_test_error() {
+#if !((HIP_VERSION_MAJOR == 5) && (HIP_VERSION_MINOR == 3))
```
Comment (dalg24) [inline core/unit_test/TestViewAPI.hpp:1545]: No, instead guard https://github.com/kokkos/kokkos/blob/4271bd9f034f028c9d93691b9506f62063f67f01/core/unit_test/TestViewAPI_d.hpp#L29-L31
and make sure you add a comment

**Review by dalg24 — CHANGES_REQUESTED** [review]

File: core/unit_test/TestViewAPI.hpp (hunk 1 above)
Comment (Rombur) [reply core/unit_test/TestViewAPI.hpp:1545]: That was for debugging. It shouldn't be there anymore

File: core/unit_test/TestViewAPI_d.hpp [hunk 2]
```diff
@@ -26,11 +26,11 @@ TEST(TEST_CATEGORY, view_api_d) {
   TestViewAPI<double, TEST_EXECSPACE>::run_test_view_operator_c();
 }
 
-// ROCm 5.3 segfaults when trying to allocate too much memory
-#if !((HIP_VERSION_MAJOR == 5) && (HIP_VERSION_MINOR == 3))
 TEST(TEST_CATEGORY, view_allocation_error) {
+#if ((HIP_VERSION_MAJOR == 5) && (HIP_VERSION_MINOR == 3))
+  GTEST_SKIP() << "ROCm 5.3 segfaults when trying to allocate too much memor";
```
Comment (dalg24) [inline core/unit_test/TestViewAPI_d.hpp:31]:
```suggestion
  GTEST_SKIP() << "ROCm 5.3 segfaults when trying to allocate too much memory";
```

**Review by dalg24 — APPROVED** [review]

**Review by crtrott — CHANGES_REQUESTED** [review]
Can we name it `ImplForceGlobalLaunch` make it clearer that it has something to do with launch mechanism?

**Comment by Rombur** [comment]
done

**Review by crtrott — APPROVED** [review]

**Comment by dalg24** [comment]
@Rombur plz cherry-pick into RC 4.0

**Pull request merged**